### PR TITLE
feat: generated Helm documentation

### LINKED ISSUE #242: Automate helm documentation update
There are some tools out there to automatically generate helm chart documentation from the values and comments. It would ensure we don't miss any entries and enforce comments in the value file.

### PR DESCRIPTION
**Pull request by aslafy-z**
Preview at https://github.com/stakater/application/blob/feat/generated-readme/README.md

- Generate README.md with https://github.com/norwoodj/helm-docs
- Add pre-commit hook and GitHub Action run to validate README
- Document the whole values file
- Move some defaults from templates to values
- Set `certificate.issuerRef.group` to `cert-manager.io`

Thanks to https://github.com/norwoodj/helm-docs/pull/260, we will soon be able to render a bit better some of the default values.

Fixes #242

### CODE REVIEW

**Comment by rasheedamir** [comment]
@karl-johan-grahn @d3adb5 can you guys review it?

**Review by karl-johan-grahn — DISMISSED** [review]

File: application/templates/alertmanagerconfig.yaml [hunk 1]
```diff
@@ -6,15 +6,15 @@ metadata:
   name: {{ template "application.name" . }}
   namespace: {{ include "application.namespace" . }}
   labels:
-  {{- include "application.labels" . | nindent 4 }}
+  {{- include "application.labels" $ | nindent 4 }}
```
Comment (SheryarButt) [inline application/templates/alertmanagerconfig.yaml:9]: Can you explain what $ does here that wasn't already covered by `.` ?

Comment (aslafy-z) [reply application/templates/alertmanagerconfig.yaml:9]: This change and some other similar ones can be reverted. I mass replaced include statements. Using $ there ensures that even if they are wrapped in a range in the future, they get th le right context.

File: .github/workflows/pull_request.yaml [hunk 2]
```diff
@@ -73,3 +73,17 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{github.event.pull_request.head.sha}}
+      - name: Run helm-docs and check diff
+        uses: losisin/helm-docs-github-action@v1
+        with:
+          output-file: README.md
+          template-files: README.md.gotmpl
+          fail-on-diff: true
```
Comment (karl-johan-grahn) [inline .github/workflows/pull_request.yaml:89]: Why should it fail on diff?

Comment (aslafy-z) [reply .github/workflows/pull_request.yaml:89]: It makes the status check fail if the values changed but the documentation was not updated in the PR

Comment (karl-johan-grahn) [reply .github/workflows/pull_request.yaml:89]: ah ok nice

**Comment by rasheedamir** [comment]
looks good to me! great work @aslafy-z

**Pull request merged**